addons/website/wizard/website_robots.py

```diff
@@ -7,8 +7,13 @@ class WebsiteRobots(models.TransientModel):
     _name = "website.robots"
     _description = "Robots.txt Editor"
 
-    content = fields.Text(default=lambda s: s.env['website'].get_current_website().robots_txt)
+    content = fields.Text()
```

Comment:
leave this untouched, just alter `website_id` in context so `get_current_website` will return the correct one (it depend of context)